spec/components/modal_component_spec.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ModalComponent, type: :component do
+  subject(:component) do
+    described_class.new(modal_id: modal_id, title: title)
+  end
+
+  let(:rendered) { render_inline(component) }
```

Comment:
I'm not sure if the component really should be the subject since you only test against `rendered`